### ISSUE TITLE
alsa-related issues

### ISSUE THREAD

**Issue by mackdroid** [nwg-piotr/nwg-panel#24]
hello there whenever i try to launch the panel i get this error
i use pipewire if that helps as it looks like its related to audio
![image](https://user-images.githubusercontent.com/68280146/109327254-23658400-787e-11eb-9f3e-bed66e51a32e.png)
anything else ill be willing to provide

**Comment by nwg-piotr** [comment]
Make sure you've got alsa-lib and alsa-utils installed. If so, please paste the output of the `amixer sget Master` command. Also: what Linux distribution do you use?

**Comment by mackdroid** [comment]
`amixer: Unable to find simple control 'Master',0`
and i have a mixer named IEC958 also i use arch linux

**Comment by nwg-piotr** [comment]
>  i use arch linux

Did you install from AUR? If so, you should have the `python-pyalsa` module installed, but it looks as if you had not.

**Comment by nwg-piotr** [comment]
Would you be able to install manually from another branch, and tell me if my fix works on your system?

**Comment by mackdroid** [comment]
Yes I do have it installed

**Comment by mackdroid** [comment]
Okay let me try now

**Comment by mackdroid** [comment]
![image](https://user-images.githubusercontent.com/68280146/109375485-1dee5500-78e3-11eb-86d5-ef8770db54ea.png)
this is the error i get now

**Comment by nwg-piotr** [comment]
First of all: please provide the output of `amixer scontrols`. It usually looks something like this:

```text
$ amixer scontrols
Simple mixer control 'Master',0
Simple mixer control 'Capture',0
```

but on your machine must be different, and I need to know details.

**Comment by nwg-piotr** [comment]
The commit above should prevent the panel from crashing on your machine, but the volume slider probably won't work. 

Clone the repository in the latest version. As a workaround:

1. Check the name that appears in your `amixer scontrols` output instead of `Master`;
2. save the name to the `scontrol` file in your nwg-panel config directory;
3. tell me if it helps.

**Comment by mackdroid** [comment]
sorry for the delay, but the panel now launches but as you mentioned the volume controls dont work

**Comment by mackdroid** [comment]
![image](https://user-images.githubusercontent.com/68280146/109668333-15be4000-7b97-11eb-9283-52b951755db8.png)
this is what i get when i try to change the volume

**Comment by nwg-piotr** [comment]
First of all: I still need to see the output of the `amixer scontrols` command. Please do type it in the terminal and show me what it prints. I can't reproduce the problem on any of my machines, so we won't take a step further w/o your help.

Secondly: have you tried to edit the `~/.local/share/nwg-config/defaults`, as I mentioned above?

**Comment by mackdroid** [comment]
Output of amixer scontrols
`Simple mixer control 'IEC958',0`

Yep and I've replaced Master with IEC958

**Comment by mackdroid** [comment]
Also do you by any chance have discord? I'm active there so you would be able to get a faster response

**Comment by nwg-piotr** [comment]
Just 1 line? Strange. Could you try adding the ~/.asoundrc file as described [here](https://bbs.archlinux.org/viewtopic.php?pid=1551549#p1551549)?

> Also do you by any chance have discord?

No, I only use Google Hangouts, And whatever we write here, will be available for future troubleshooters.

**Comment by nwg-piotr** [comment]
Sorry, it got closed automatically on the latest merge.

It turns out I forgot to change the [path](https://github.com/nwg-piotr/nwg-panel/pull/29/commits/78881233d53014d115f1179ea198844fb3a7c300). Could you try installing manually once again?

**Comment by mackdroid** [comment]
alright
yea i only have one line 
![image](https://user-images.githubusercontent.com/68280146/109748869-9ca70300-7bff-11eb-9762-09de3a8ba33b.png)

**Comment by mackdroid** [comment]
![image](https://user-images.githubusercontent.com/68280146/109749007-dbd55400-7bff-11eb-9c2a-658b7d969d95.png)
well the volume slider does not work and i get this instead now

**Comment by mackdroid** [comment]
also you can use pamixer if amixer doesnt work 
it works great with pipewire

**Comment by nwg-piotr** [comment]
> also you can use pamixer if amixer doesnt work

This would be the 3rd method, as we now use `python-psutils` if available, and `amixer` as fallback. It doesn't make sense to do it this way. 

I need to check if we could rely on `pamixer` as the only method. Would you be able to clone and install from another branch, in case it turns out to work for me?

**Comment by mackdroid** [comment]
Sure

**Comment by nwg-piotr** [comment]
Alright, it works, at least on my old testing laptop. Can't test it better here, it must wait till I'm home. Take a look, please.

```text
git clone https://github.com/nwg-piotr/nwg-panel.git -b alsa
cd nwg-panel
sudo ./install.sh
```

**Comment by mackdroid** [comment]
alrighty

**Comment by mackdroid** [comment]
`Traceback (most recent call last):
  File "/usr/bin/nwg-panel", line 33, in <module>
    sys.exit(load_entry_point('nwg-panel==0.1.6', 'gui_scripts', 'nwg-panel')())
  File "/usr/bin/nwg-panel", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.9/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 790, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/usr/lib/python3.9/site-packages/nwg_panel-0.1.6-py3.9.egg/nwg_panel/main.py", line 29, in <module>
    from nwg_panel.tools import *
  File "/usr/lib/python3.9/site-packages/nwg_panel-0.1.6-py3.9.egg/nwg_panel/tools.py", line 18, in <module>
    import nwg_panel.common
  File "/usr/lib/python3.9/site-packages/nwg_panel-0.1.6-py3.9.egg/nwg_panel/common.py", line 7, in <module>
    from i3ipc import Connection
ModuleNotFoundError: No module named 'i3ipc'` i get this error

**Comment by mackdroid** [comment]
oh wait never mind im dumb its a missing module

**Comment by mackdroid** [comment]
the panel does launch but now the brightness controls dont work, i do have light installed and it says this 
`Couldn't set brightness, is 'light' installed?`

**Comment by nwg-piotr** [comment]
Yep, see [dependencies](https://github.com/nwg-piotr/nwg-panel/wiki/Installation#dependencies).

**Comment by mackdroid** [comment]
but i do have light installed
![image](https://user-images.githubusercontent.com/68280146/109813080-d9e8b080-7c52-11eb-9f73-1c494669e2e4.png)
and brightness controls used to work

**Comment by nwg-piotr** [comment]
Check if this returns `0` on your machine:

```text
$ light -S 20
$ echo $?
0
```

**Comment by nwg-piotr** [comment]
One way or another, you may clone again. I did quite a massive code clearance on the `alsa` branch.

**Comment by nwg-piotr** [comment]
Resolved in v0.1.7
